src/3D/setup3d.F90

```diff
@@ -799,6 +807,11 @@ subroutine setup(fileName, fileType)
             end do
         end do patchLoop
 
+        ! if there was an error in the BC setup stop
+        if (bcError) then
+            stop
```

Comment:
I realize that this was like this before, but since this is a parallel code, we should probably call `mpi_abort` to exit more gracefully. However, since there are other locations, we could also defer this to a separate PR.

Reply:
There are several places where the error handling isn't done properly in parallel and MPI gives wild errors, I could make an issue to address that.